shapefile.py

```diff
@@ -939,10 +941,19 @@ def poly(self, parts=[], shapeType=POLYGON, partTypes=[]):
             polyShape.partTypes = partTypes
         self._shapes.append(polyShape)
 
-    def field(self, name, fieldType="C", size="50", decimal=0):
+    def field(self, name, fieldType='C', size=50, decimal=0):
         """Adds a dbf field descriptor to the shapefile."""
+        fieldType = fieldType.upper()
+        size = int(size)
+        decimal = int(decimal)
+        if fieldType == 'D' and size < 8:
+            raise ShapefileException("Date fields require min size of 8 (%d is too small)" % size)

```

Comment:
What happens when you have a date with size > 8? In my tests it seems to result in corrupt shapefiles. If I save a date with a size bigger than 8, then open the shapefile in arcmap and try to edit it, the data in the dbf file become corrupted. 

Might it be better to mandate a size of exactly 8?